fuseimg/datasets/ukbb_neck_to_knee.py

```diff
@@ -210,11 +213,15 @@ def dataset(
         :param is_female                    filter only male/females from database
         :return: DatasetDefault object
         """
-        if is_female == None:
-            all_sample_ids = [file for file in os.listdir(data_dir) if '.zip' in file]
-        else:
-            all_sample_ids = [file for file in os.listdir(data_dir) if '.zip' in file]
-            all_sample_ids = list(set(input_source_gt[input_source_gt['is female'] == is_female][input_source_gt['file'].isin(all_sample_ids)]['file'].to_list()))
+
+        existing_files = [file for file in os.listdir(data_dir) if '.zip' in file]
+        existing_sample_id_fields = [f.split("_") for f in existing_files]
+        existing_sample_ids = set([a[0] + "_*_" + a[2] + "_" + a[3] for a in existing_sample_id_fields])
+        a_filter = input_source_gt['file'].isin(existing_sample_ids)
+        if is_female is not None:
+            a_filter &= input_source_gt['is female'] == is_female
```

Comment:
great idea on how to merge filter conditions .
if we have more complex situations ( like doing or or negation ) should we support?

Reply:
I am changing this code  - to make it more generic.